android/src/main/kotlin/com/example/plug_in_kotlin/PlugInKotlinPlugin.kt

```diff
@@ -1,36 +1,98 @@
 package com.example.plug_in_kotlin
 
+import android.Manifest
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
 import androidx.annotation.NonNull
-
+import androidx.core.app.ActivityCompat
 import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
 import io.flutter.plugin.common.MethodCall
 import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
-import io.flutter.plugin.common.MethodChannel.Result
-import io.flutter.plugin.common.PluginRegistry.Registrar
-
-/** PlugInKotlinPlugin */
-class PlugInKotlinPlugin: FlutterPlugin, MethodCallHandler {
-  /// The MethodChannel that will the communication between Flutter and native Android
-  ///
-  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-  /// when the Flutter Engine is detached from the Activity
-  private lateinit var channel : MethodChannel
-
-  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "plug_in_kotlin")
-    channel.setMethodCallHandler(this)
-  }
-
-  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
-    } else {
-      result.notImplemented()
-    }
-  }
-
-  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
-    channel.setMethodCallHandler(null)
-  }
+import io.flutter.plugin.common.PluginRegistry.ActivityResultListener
+
+class PlugInKotlinPlugin : FlutterPlugin, MethodCallHandler, ActivityAware, ActivityResultListener {
+    private lateinit var channel: MethodChannel
+
+    private var context: Context? = null
+    private var activity: Activity? = null
+
+    companion object {
+        private const val REQUEST_ACCESS_FINE = 0
+    }
+
+    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        context = flutterPluginBinding.applicationContext
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "plug_in")
+        channel.setMethodCallHandler(this)
+    }
+
+    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
+        when (call.method) {
+            "getPlatformVersion" -> result.success("Android " + Build.VERSION.RELEASE)
+            "initializeLocatorPlugin" -> initializeLocatorPlugin(result)
+            "checkPermissions" -> checkPermissions(result)
+            "requestPermissions" -> requestPermissions()
```

Comment:
Same comment I made in a previous PR to another dev, we need to return the control to Flutter with the result, if not, a memory leak can occur, because a string reference to the `Future` in Flutter

Reply:
You are referring to the requestPermissions method, right?

Reply:
done